pkg/detectors/twilio/twilio.go

```diff
@@ -73,10 +85,18 @@ func (s Scanner) FromData(ctx context.Context, verify bool, data []byte) (result
 				req.SetBasicAuth(sid, key)
 				res, err := client.Do(req)
 				if err == nil {
-					res.Body.Close() // The request body is unused.
+					defer res.Body.Close()
+					var serviceResponse serviceResponse
+					if err := json.NewDecoder(res.Body).Decode(&serviceResponse); err != nil {
+						err = fmt.Errorf("failed to decode service response: %w", err)
+						s1.SetVerificationError(err, key)
```

Comment:
"Verification errors" are errors that interfered with TruffleHog's ability to determine whether a secret is verified, so they should only be set when TruffleHog was unable to do that. In particular, `SetVerificationError` shouldn't be called on results with `Verified == true`, which would be a possibility after your change.

This does mean that there's not a good way to store non-fatal errors within a result structure, which is admittedly unfortunate, but it's the reality right now. In this particular case I believe your two choices if this decoding fails are:
1. Assume that verification failed (because the endpoint is probably behaving unexpectedly), even if the HTTP response is 2xx. In that case you would just `continue` here to skip the actual HTTP status check.
1. Just ignore it - the important part is whether the HTTP response indicates a verified credential, not this additional metadata.

I think that either position is defensible. Personally I lean towards (2) but I don't speak with authority on that.

Reply:
@rgmz do you have any good examples you can provide? The invariant exists because it simplifies the underlying logical model and we've lacked concrete cases that would justify adding more complexity. But if you've got some that calculus might change.

Reply:
In case of Twilio, I would also lean towards (2) as well because originally the verification is dependent on Status.OK not on the response. Additional Metadata is optional, I believe we can skip it if there is error in parsing response or reading the stream.

Reply:
Btw, I found the similar issue in Slack detector. I will create a separate PR for that.